modules/aws_s3/tf_module/bucket.tf

```
@@ -96,7 +96,7 @@ resource "aws_cloudfront_origin_access_identity" "read" {
 
 
 data "aws_iam_policy_document" "s3_policy" {
-  source_json = var.bucket_policy == null ? "" : jsonencode(var.bucket_policy)
+  source_policy_documents = var.bucket_policy == null ? [] : jsonencode(var.bucket_policy)
```

```suggestion
  source_policy_documents = var.bucket_policy == null ? [] : [jsonencode(var.bucket_policy)]
```

`source_policy_documents` is supposed to be a list of documents, so I think we should wrap the value returned from `jsonencode`